src/common/logging.cpp

```diff
@@ -28,64 +28,270 @@
 
 #include "logging.hpp"
 
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
 #include <assert.h>
 #include <errno.h>
 #include <stdarg.h>
 #include <stdint.h>
 #include <string.h>
 #include <syslog.h>
+#include <sys/time.h>
 
 static int        sLevel = LOG_INFO;
 static const char HEX_CHARS[] = "0123456789abcdef";
 
+static unsigned msecs_start;
```

Comment:
I thought this was for public functions only, not private functions - that said I can  change this.